### PR TITLE
Cocoapods: Also port the lockfile parsing to KxS

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

### CODE REVIEW

File: plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt [hunk 1]
```diff
@@ -48,40 +53,45 @@ internal data class Lockfile(
 }
 
 internal fun String.parseLockfile(): Lockfile {
-    val root = yamlMapper.readTree(this)
+    val root = Yaml.default.parseToYamlNode(this).yamlMap
+
+    val pods = root.yamlMap.get<YamlList>("PODS")!!.yamlList.items.map { node ->
```
Comment (sschuberth) [inline plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt]: You already call `yamlMap` as part of defining `root`, so it can be dropped here.

File: plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt [hunk 2]
```diff
@@ -48,40 +53,45 @@ internal data class Lockfile(
...
             else -> {
-                val (name, version) = parseNameAndVersion(node.textValue())
+                val (name, version) = parseNameAndVersion(node.yamlScalar.content)
                 Lockfile.Pod(name, version)
             }
         }
     }
 
-    val checkoutOptions = root.get("CHECKOUT OPTIONS")?.fields()?.asSequence()?.mapNotNull { (name, node) ->
+    val checkoutOptions = root.yamlMap.get<YamlMap>("CHECKOUT OPTIONS")?.entries.orEmpty().mapNotNull {
```
Comment (sschuberth) [inline plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt]: You already call `yamlMap` as part of defining `root`, so it can be dropped here.

File: plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt [hunk 3]
```diff
@@ -48,40 +53,45 @@ internal data class Lockfile(
...
+            commit = node.get<YamlScalar>(":commit")?.content.orEmpty()
         )
 
         name to checkoutOption
-    }.orEmpty().toMap()
+    }.toMap()
 
-    val dependencies = root.get("DEPENDENCIES").map { node ->
-        val (name, version) = parseNameAndVersion(node.textValue())
+    val dependencies = root.yamlMap.get<YamlList>("DEPENDENCIES")?.items.orEmpty().map { node ->
```
Comment (sschuberth) [inline plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt]: You already call `yamlMap` as part of defining `root`, so it can be dropped here.

File: plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt [hunk 4]
```diff
@@ -48,40 +53,45 @@ internal data class Lockfile(
 }
 
 internal fun String.parseLockfile(): Lockfile {
-    val root = yamlMapper.readTree(this)
+    val root = Yaml.default.parseToYamlNode(this).yamlMap
+
+    val pods = root.yamlMap.get<YamlList>("PODS")!!.yamlList.items.map { node ->
+        when {
+            node is YamlMap -> {
+                val mapEntry = node.yamlMap.entries.entries.single()
```
Comment (sschuberth) [inline plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt]: Maybe use `val (key, value) =` here?

File: plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt (hunk 3 above)
Comment (sschuberth) [reply plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt]: While I'm not a fan of using `!!`, the previous code assumed `get("DEPENDENCIES")` to always be non-null, so we probably use `!!` here to document that like you also did in line 58.

File: plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt [hunk 5]
```diff
@@ -87,10 +87,7 @@ private fun YamlNode.toPod(): Pod =
             Pod(
                 name = name,
                 version = version,
-                dependencies = mapEntry.value.yamlList.items.map {
-                    val (depName, depVersion) = parseNameAndVersion(it.yamlScalar.content)
-                    Pod(depName, depVersion)
-                }
+                dependencies = mapEntry.value.yamlList.items.map { it.toPod() }
```
Comment (sschuberth) [inline plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt:90]: Nit: As it's a bit hard to see from the collapsed diff, maybe explain in the commit message that this is calling `toPod()` recursively, assuming that all list entries are scalars.

File: plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt [hunk 6]
```diff
@@ -56,7 +56,7 @@ internal data class Lockfile(
 
 internal fun String.parseLockfile(): Lockfile {
     val root = Yaml.default.parseToYamlNode(this).yamlMap
-    val pods = root.get<YamlList>("PODS")!!.items.map { it.toPod() }
+    val pods = root.get<YamlList>("PODS")?.items.orEmpty().map { it.toPod() }
```
Comment (sschuberth) [inline plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt:59]: This relates a bit to my comment on the first commit. While I'm fine with the change per se, I'd appreciate if this semantic change could already be done as a preparation on the existing code, before the migration, and then have no more semantic changes as part of the migration.

File: plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt [hunk 7]
```diff
@@ -58,7 +58,7 @@ internal fun String.parseLockfile(): Lockfile {
     val root = Yaml.default.parseToYamlNode(this).yamlMap
     val pods = root.get<YamlList>("PODS")?.items.orEmpty().map { it.toPod() }
 
-    val checkoutOptions = root.get<YamlMap>("CHECKOUT OPTIONS")?.entries.orEmpty().mapNotNull {
+    val checkoutOptions = root.get<YamlMap>("CHECKOUT OPTIONS")?.entries.orEmpty().map {
```
Comment (sschuberth) [inline plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt:61]: Could this as well please be done before migration, as one of the first commits in this PR, to better explain that it is unrelated to be migration?

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt (hunk 6 above)
Comment (fviernau) [reply plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt:59]: I tried, but moving it before the migration looks a bit ugly to me. 
Also, to me the added value is not obvious. 
I can offer to either keep it here or drop it.

**Review by sschuberth — APPROVED** [review]
